File: app/src/main/java/me/kr328/nevo/decorators/smscaptcha/SettingsReceiver.java
package me.kr328.nevo.decorators.smscaptcha;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class SettingsReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        context.startActivity(new Intent(context ,SettingsActivity.class));
    }
}

File: app/src/main/java/me/kr328/nevo/decorators/smscaptcha/SettingsReceiver.kt
package me.kr328.nevo.decorators.smscaptcha

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class SettingsReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        context.startActivity(Intent(context, SettingsActivity::class.java)
                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK))
    }
}